sn_auditor/src/dag_db.rs

```diff
@@ -381,7 +381,12 @@ impl SpendDagDb {
                 } else {
                     BTreeSet::new()
                 };
-                let _ = beta_tracking.forwarded_payments.insert(p, payments);
+                if beta_tracking.forwarded_payments.get(&p).is_none() {
+                    let existing = beta_tracking.forwarded_payments.insert(p.clone(), payments);
+                    if existing.is_some() {
```

Comment:
better do
```
if let Some(existing) = beta_tracking.forwarded_payments.insert(p.clone(), payments) {
     warn!("Overwriting existing participant {p} with new participant {hash:?}");
    let _ = beta_tracking.forwarded_payments.insert(p.clone(), payments.extend(existing));
}
```
to avoid lost previous count